Optimize guest code

energyGiver opened the pull request:
This PR closes issue #3 

# Changes
- Remove Unnecessary variables and logics in the guset code
- Apply succinct receipt(recursive zk stark proof + zk snark)
- Benchmark based on the type of receipt.


  | Composite Receipt | Fast Receipt | Succinct Receipt
-- | -- | -- | --
proving method | zk stark | zk snark | zk stark + snark
Hash function | Poseidon2 | sha256 | Poseidon2
Proving time | 126.383148166s | **114.381178625s** | 154.6101965s
Verifying time | 123.834666ms | 18.826084ms | **10.545125ms**

energyGiver committed:
Optimize guest code

energyGiver committed:
Change input and output of guest

woojucy commented:
In the host side code, the `input` is being set, and this `input` is then accessed from the guest side code. I believe that during the verification process, the input should be reflected in the `receipt` object. It would be helpful if this part could be clarified and properly reflected in the code.

energyGiver committed:
Add an exponent to the input of guest

energyGiver committed:
Chagne risc0 version and guest code

energyGiver commented:
> In the host side code, the `input` is being set, and this `input` is then accessed from the guest side code. I believe that during the verification process, the input should be reflected in the `receipt` object. It would be helpful if this part could be clarified and properly reflected in the code.

@woojucy Thank you for your comment. According to https://docs.rs/risc0-zkvm/2.0.1/risc0_zkvm/guest/index.html, public input is set using env::commit, and other inputs are private.

I modified exponent as an input, and changed the public output like this:
env::commit(&(base, modulus, range, calculation));

woojucy commented:
Got it! Thanks.
